### ISSUE TITLE
shared file download link bypasses password!

### ISSUE THREAD

**Issue by JustinWebDev** [filebrowser/filebrowser#3470]
**Description**
shared file download link bypasses password

**Expected behaviour**
the password should be requested before the download starts

**What is happening instead?**
the user is not prompted for a password

**Additional context**

**How to reproduce?**
share a file > set a password for the share
copy the download link for the share, e.g.:
http://192.168.0.11:8080/api/public/dl/NAe9rn14/21.mp4?token=VW-XkFrTvZxAJ3mp5c0InBcc74JObXsN5FZrYmlE8P2X8S-B_DcD-YsfossKDtGwFAR1l2pozfH-P4nppca-718Ssr28qFyoPWEAZliF6cZLkcg4NyrEG3EF3sIDyf_P

on another machine, browse to the download link - observe the browser starts to download and does not prompt for the password

**Files**

**Comment by JustinWebDev** [comment]
Why are the devs ignoring this security issue?

**Comment by didotb** [comment]
I don't think this is a security issue, more of a frontend alt text issue.
There should be 2 buttons to share the file, the second icon of a clipboard with an arrow should be a `Copy public shareable link`, which should disable the password prompt as the link provides a token already.

**Comment by JustinWebDev** [comment]
So it is a security-related bug.  The frontend bug makes you think you have a protected link when you don't.  It may not be a vulnerability, but it is a security risk for certain and a user won't be aware of it.

The first copy icon says 'copy to clipboard' which is ambiguous in this case.  Why don't you just explicitly say "Copy password-protected link" like you are saying for the public link.  No reason not to be clear.  Isn't this like a 1-minute fix?

Why is there even a public link after you password protect it?  There is the security failure.  And, since the file can have multiple shares against it, if the share has a password, don't even offer the public link, because if a public link is needed, then a new share can be created w/o a password.

**Comment by JustinWebDev** [comment]
This one should never go stale.

**Comment by JustinWebDev** [comment]
This one should never go stale.

Thank you, bot!

**Comment by didotb** [comment]
> This one should never go stale.
> 
> Thank you, bot!

You'll never get anywhere by doing it like this, how about make your own fork and fix it yourself?

**Comment by lordpayder** [comment]
> > This one should never go stale.
> > Thank you, bot!
> 
> You'll never get anywhere by doing it like this, how about make your own fork and fix it yourself?

I'm sure he would have already done that if he were familiar with the relevant programming languages.

The fact is, however, that there’s a problem here because additional security is assumed, but it doesn’t work correctly.
Either it’s as you say and this is just a "frontend alt text issue," or the password function doesn’t work properly for individual files.

Personally, I share links with passwords via separate channels, so I’d appreciate a fix. However, this probably doesn’t apply to the majority of users. My current workaround is to only share folders.

**Comment by JustinWebDev** [comment]
Yeah, I'm kinda bewildered by the response.  If you expect everyone to fix this code for various issues, then why report issues at all?

If I even knew where to start, and it is as superficial as a text string, I might be able to work my way thru Go.  But maintainers can't assume every user is also a programmer suitable for the project.

**Comment by JustinWebDev** [comment]
it should be fresh
